akka-cluster-tools/src/multi-jvm/scala/akka/cluster/client/ClusterClientSpec.scala

```diff
@@ -63,6 +62,115 @@ object ClusterClientSpec extends MultiNodeConfig {
     }
   }
 
+  //#clientEventsListener
+  class ClientListener(targetClient: ActorRef) extends Actor {
+    override def preStart(): Unit = {
+      context.system.eventStream.subscribe(self, classOf[ContactPointChange])
+      targetClient ! GetContactPoints
+    }
+
+    def receive: Receive =
+      starting(Set.empty[ActorSelection])
+
+    def starting(contactPoints: Set[ActorSelection]): Receive = {
+      case ContactPointAdded(cp, `targetClient`) ⇒
+        context.become(starting(contactPoints + cp))
+      case ContactPointRemoved(cp, `targetClient`) ⇒
+        context.become(starting(contactPoints - cp))
+      case ContactPoints(cps) ⇒
+        context.become(started(contactPoints ++ cps))
```

Comment:
> On 9 May 2016, at 9:07 PM, Patrik Nordwall <notifications@github.com> wrote:
> 
> ok, I don't know what you mean and it probably doesn't matter
> 
> 
It matters to me… and here’s why...
> note that there are no async boundaries between publishing to event stream and enqueuing the message to the subscriber's mailbox
> 
> 
I never realised that. I had just assumed that the event bus was an actor in its own right. This does indeed change (simplify) things.

Thanks.